tests/run-make/llvm-ident/rmake.rs

```diff
@@ -0,0 +1,40 @@
+//@ only-linux
+
+use run_make_support::llvm::llvm_bin_dir;
+use run_make_support::{cmd, env_var, llvm_filecheck, read_dir, rustc, source_root};
+
+use std::ffi::OsStr;
+
+fn main() {
+    // `-Ccodegen-units=16 -Copt-level=2` is used here to trigger thin LTO
+    // across codegen units to test deduplication of the named metadata
+    // (see `LLVMRustPrepareThinLTOImport` for details).
+    rustc()
+        .emit("link,obj")
+        .arg("-Csave-temps")
+        .codegen_units(16)
+        .opt_level("2")
+        .target(&env_var("TARGET"))
+        .arg("-")
+        .stdin("fn main(){}")
+        .run();
+
+    // `llvm-dis` is used here since `--emit=llvm-ir` does not emit LLVM IR
+    // for temporary outputs.
+    let mut files = Vec::new();
+    read_dir(".", |path| {
+        if path.is_file() && path.extension().is_some_and(|ext| ext == OsStr::new("bc")) {
+            files.push(path.to_path_buf());
+        }
+    });
+    cmd(llvm_bin_dir().join("llvm-dis")).args(&files).run();
+
+    // Check LLVM IR files (including temporary outputs) have `!llvm.ident`
+    // named metadata, reusing the related codegen test.
+    let llvm_ident_path = source_root().join("tests/codegen/llvm-ident.rs");
```

Comment:
Might be nice to eventually introduce helpers like `for_each_file(|p| has_extension(p, "ll"), |path| { ... })` and `get_files_with_ext(dir, ".ext")` to `run-make-support`, because the `*.ext` bash patterns repeat in multiple tests. But it doesn't need to be in this PR.